core/src/androidMain/kotlin/io/github/aag/core/data/background/UpdateAuthTokenWorker.kt

```diff
@@ -0,0 +1,42 @@
+package io.github.aag.core.data.background
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import io.github.aag.core.domain.models.UserProfile
+import io.github.aag.core.domain.repositories.AuthRepository
+import io.github.aag.core.domain.repositories.ProfileRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class UpdateAuthTokenWorker(
+    appContext: Context,
+    workerParams: WorkerParameters,
+    private val authRepository: AuthRepository,
+    private val useProfileRepository: ProfileRepository
+) : CoroutineWorker(appContext = appContext, params = workerParams) {
+    private var userProfile: UserProfile? = null
+
+    init {
+        CoroutineScope(Dispatchers.IO + Job())
```

Comment:
`coroutineContext` of `CoroutineWorker` is deprecated, I wonder what's better to use here 🤔 It runs once, I don't need to cancel it as Worker runs periodically and it should always has up-to-date `refreshToken` (use may log out and log back in, e.g.)

Reply:
You shouldn't create additional scopes inside `CoroutineWorker` and you shouldn't call `doWork` manually

Reply:
Removed `doWork()` call and creation of a new scope. Used lifecycleScope instead, though it is Application's lifecycle :)